### PR TITLE
Improve lambda list indentation

### PR DESCRIPTION
**Pull request by galdor**
These patches should fix the issue reported in #744.

More comments in commit messages.

### CODE REVIEW

File: contrib/slime-cl-indent.el [hunk 1]
```diff
@@ -1729,6 +1729,7 @@ Cause subsequent clauses to be indented.")
              (defconstant (as defvar))
              (defcustom   (4 2 2 2))
              (defparameter     (as defvar))
+             (define-compiler-macro (as defun))
```
Comment (luismbo) [inline contrib/slime-cl-indent.el:1744]: I think these are supposed to be sorted alphabetically. If you find others that are out of order and want to sort them please make that a different commit.

Comment (galdor) [reply contrib/slime-cl-indent.el:1743]: Done

File: contrib/slime-cl-indent.el [hunk 2]
```diff
@@ -1116,14 +1120,22 @@ environment\\|more\
...
-                        next nil)
-                (setq indent col))))
+              (cond
+               ((looking-at lisp-indent-lambda-list-single-arg-keywords-regexp)
+                ;; Some keywords such as &whole have a single argument;
+                ;; following arguments are indented to the beginning of the
+                ;; lambda-list.
+                (setq indent col
+                      next nil))
+               ((looking-at lisp-indent-lambda-list-keywords-regexp)
```
Comment (luismbo) [inline contrib/slime-cl-indent.el:1130]: There's another reference to `lisp-indent-lambda-list-keywords-regexp` earlier in this function. Does it need to be adapted too?

Comment (galdor) [reply contrib/slime-cl-indent.el:1130]: I do not think so; I tried lots of adjustments because the function is really opaque, and this is how I got indentation which works as intended in without affecting other lambda lists.

I'm pretty sure that there are other edge cases (e.g. the fact that &ALLOW-OTHER-KEYS does not take any argument), but at this point I am just trying to fix a very specific problem without breaking anything.

Comment (luismbo) [reply contrib/slime-cl-indent.el:1130]: ACK. Sounds reasonable.

**Comment by luismbo** [comment]
Thanks.
